### PR TITLE
Use git credentials in packer_pipeline

### PR DESCRIPTION
**Pull request by knightsamar**
This will be used for repo checkouts in `build_image.sh` as well as status updates on the packer-templates PR

### CODE REVIEW

**Comment by knightsamar** [comment]
@doublej472 the credentials will be taken from the data bag (a MR for that was approved today) and put in the `.git-credentials` file for use by berks when it tries to provision an image (eg. the mitaka-aio and mitaka-identity images in osuosl/packer-templates#30)

The same credentials are also utilized by the pipeline in Jenkinsfile [when setting commit status on a PR](https://github.com/osuosl/packer-templates/pull/30/commits/d022e774b541155b594072de7eba6e9f643d44da#diff-58231b16fdee45a03a4ee3cf94a9f2c3R90)

**Comment by knightsamar** [comment]
This PR is actually *now* ready for review!

File: spec/unit/recipes/packer_pipeline_node.rb [hunk 1]
```diff
@@ -78,6 +78,13 @@
       end
 
       it do
+        expect(chef_run).to create_template('/home/alfred/.git-credentials').with(
+          user: 'alfred',
+          group: 'alfred'
```
Comment (ramereth) [inline spec/unit/recipes/packer_pipeline_node.rb]: You should also test the variables parameter here.

File: test/integration/packer_pipeline_node/serverspec/packer_pipeline_node_spec.rb [hunk 2]
```diff
@@ -63,3 +63,8 @@
 describe command('berks version') do
   its(:exit_status) { should eq 0 }
 end
+
+describe file('/home/alfred/.git-credentials') do
+  it { should be_a_file }
+  its(:content) { should match('https://osuosl-manatee') }
```
Comment (ramereth) [inline test/integration/packer_pipeline_node/serverspec/packer_pipeline_node_spec.rb]: I would check for the whole line including the token to be sure it works properly.

**Comment by ramereth** [comment]
Actually, I would make the ``/home/alfred/.git-credentials`` file with mode 600 to be safe.

**Comment by ramereth** [comment]
@knightsamar your ChefSpec files need to be named with ``_spec.rb`` on the end. They aren't even being tested :(

**Comment by knightsamar** [comment]
@ramereth ready for a final review! Let me know and I will squash it and merge it.

**Comment by knightsamar** [comment]
@ramereth squashed.

**Comment by ramereth** [comment]
!bump patch *

**Comment by osuosl-manatee** [comment]
Jenkins has merged this PR into `master` and has automatically performed a patch-level version bump to v1.8.14.  Have a nice day!

**Pull request merged**